vocabularies/UI.xml

```diff
@@ -1401,6 +1401,37 @@ The trend is
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
+      <Term Name="InputMask" Type="UI.InputMaskType" Nullable="false" AppliesTo="Property Parameter">
+        <Annotation Term="Core.Description" String="Properties or parameters annotated with this term will get a mask in edit mode." />
+        <Annotation Term="Core.LongDescription">
+          <String>Input masks improve readability and help to enter data correctly. 
+So, masks can be especially useful for input fields that have a fixed pattern, e.g. DUNS numbers or similar. 
+[Here](../examples/UI.InputMask-sample.xml) you can find an example for this annotation</String>
+        </Annotation>
+      </Term>
+      <ComplexType Name="InputMaskType">
+        <Property Name="Mask" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.Description" String="The mask to be applied to the property or the parameter" />
+        </Property>
+        <Property Name="Placeholder" Type="Edm.String" DefaultValue="_">
+          <Annotation Term="Core.Description" String="A single character symbol to be shown where the user can type a character" />
+        </Property>
+        <Property Name="InputMaskRules" Type="Collection(UI.InputMaskRuleType)" Nullable="false">
```

Comment:
Thanks, now please add an example using this

Reply:
I like the example and wonder whether we can keep it Germany-centric or whether we should generalize it according to https://en.wikipedia.org/wiki/International_Bank_Account_Number

> The IBAN consists of up to 34 alphanumeric characters, as follows:
>
> - country code using [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) – two letters,
> - [check digits](https://en.wikipedia.org/wiki/Check_digit) – two digits, and
> - **Basic Bank Account Number (BBAN)** – up to 30 alphanumeric characters that are country-specific.[[2]](https://en.wikipedia.org/wiki/International_Bank_Account_Number#cite_note-IBANRegistry-2)

Reply:
https://experience.sap.com/internal/fiori-design-web/generic-mask-input/#do-not-use-mask-input-if

Reply:
> Is the control then expected to convert lower case letters to upper case?

yes, it will adapt the input to the mask.

Reply:
Does it parse the regular expression to know that?

Reply:
> The IBAN consists of up to 34 alphanumeric characters, as follows:

An input mask makes sense only if the length of the input is fixed, not "up to N characters".

Reply:
> An input mask makes sense only if the length of the input is fixed, not "up to N characters".

exactly, that's basically what the Fiori Design Guideline says about it.

Reply:
> Does it parse the regular expression to know that?

I have to correct myself - apparently it prevents the input of lower case letters: https://sapui5.hana.ondemand.com/#/entity/sap.m.MaskInput/sample/sap.m.sample.MaskInput. Honestly, I personally would expect a correction from lower to upper case.

Reply:
> I personally would expect a correction from lower to upper case.

But this is not expressed in the `InputMask` annotation.

Reply:
Alternative example:
```json
"SwiftCode@UI.Placeholder": "Has the format `Bank Country Location Branch`",
"SwiftCode@UI.InputMask": {
  "Mask": "CCCC CC AA AAA",
  "Placeholder": ".",
  "Rules": [
    { "MaskSymbol": "C", "RegExp": "[A-Z]" },
    { "MaskSymbol": "A", "RegExp": "[A-Z0-9]" }
  ]
}
```

Reply:
> > I personally would expect a correction from lower to upper case.
> 
> But this is not expressed in the `InputMask` annotation.

I think that's something that the client control has to decide, how it deals with that annotation and the defined rules. Neither I would exclude such a behavior nor do I want to enforce it. It should do something meaningful that the end user expects (something that UX Designer would specify usually).

Reply:
> Alternative example:

this one looks OK to me - the app could fill the last 3 characters if the user doesn't provide it.

Reply:
> the client control has to decide, how it deals with that annotation and the defined rules

If we expect the client to understand the regular expressions, how does this differ from a [`Validation.Pattern`](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Validation.V1.md#Pattern) annotation? The latter is more complex, because our regular expressions only describe one character at a time. But does that justify introducing this separate term?

Reply:
Hmm difficult, I think a general idea (at least on my side) was, to ease the definition of this annotation by having lots of defaults that come out-of-the-box.

Reply:
But this does not help if these defaults can be overwritten, see https://github.com/SAP/odata-vocabularies/pull/317#discussion_r1593961731.

Reply:
I wanted to ease it for the (app) developer :-) - not the client code.